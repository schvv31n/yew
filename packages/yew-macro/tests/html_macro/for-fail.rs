fn main() {
    _ = ::yew::html!{for x};
    _ = ::yew::html!{for x in};
    _ = ::yew::html!{for x in 0 .. 10};
    _ = ::yew::html!{for (x, y) in 0 .. 10 {
        <span>{x}</span>
    }};
}
