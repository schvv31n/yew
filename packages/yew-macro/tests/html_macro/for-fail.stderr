error: unexpected end of input, expected `in`
 --> tests/html_macro/for-fail.rs:2:9
  |
2 |     _ = ::yew::html!{for x};
  |         ^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `::yew::html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected an expression
 --> tests/html_macro/for-fail.rs:3:9
  |
3 |     _ = ::yew::html!{for x in};
  |         ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `::yew::html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected curly braces
 --> tests/html_macro/for-fail.rs:4:9
  |
4 |     _ = ::yew::html!{for x in 0 .. 10};
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `::yew::html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: diverging expression in the body of a for loop
       `break` or `continue` are not allowed in `html!` for loops
 --> tests/html_macro/for-fail.rs:9:16
  |
9 |         <span>{break}</span>
  |                ^^^^^

error[E0308]: mismatched types
 --> tests/html_macro/for-fail.rs:5:26
  |
5 |     _ = ::yew::html!{for (x, y) in 0 .. 10 {
  |                          ^^^^^^    ------- this is an iterator with items of type `{integer}`
  |                          |
  |                          expected integer, found tuple
  |
  = note: expected type `{integer}`
            found tuple `(_, _)`
